src/main.rs

```diff
@@ -40,10 +41,34 @@ struct StateRow {
     event_id: String,
 }
 
+fn parse_request_uri(req_uri: RequestUri) -> hyper::Url {
+    match req_uri {
+        RequestUri::AbsolutePath(s) => {
+            // ffs
+            return hyper::Url::parse(&format!("http://foo{}", s)).unwrap();
```

Comment:
You can also just have implicit returns here and drop the `return` and semi colon.